README.md

```diff
@@ -727,6 +727,167 @@ Vue.component('courses-list', {
 
 :arrow_right: https://docs.angularjs.org/api/ng/service/$compile
 
+### Angular
+[Live Examples Here!](https://angular.io/generated/live-examples/lifecycle-hooks/eplnkr.html)
+
+#### [ngOnChanges()](https://angular.io/guide/lifecycle-hooks#onchanges)
+Respond when Angular (re)sets data-bound input properties. The method receives a `SimpleChanges` object of current and previous property values.
+
+Called before `ngOnInit()` and whenever one or more data-bound input properties change.
+```typescript
+export class PeekABooComponent extends PeekABoo implements OnChanges {
+
+  // ...
+  
+  // only called for/if there is an @input variable set by parent.
+  ngOnChanges(changes: SimpleChanges) {
+    let changesMsgs: string[] = [];
+    for (let propName in changes) {
+      if (propName === 'name') {
+        let name = changes['name'].currentValue;
+        changesMsgs.push(`name ${this.verb} to "${name}"`);
+      } else {
+        changesMsgs.push(propName + ' ' + this.verb);
+      }
+    }
+    this.logIt(`OnChanges: ${changesMsgs.join('; ')}`);
+    this.verb = 'changed'; // next time it will be a change
+  }
+  
+  // ...
+  
+}
+```
+
+#### [ngOnInit()](https://angular.io/guide/lifecycle-hooks#oninit)
+Initialize the directive/component after Angular first displays the data-bound properties and sets the directive/component's input properties.
+
+Called once, after the first `ngOnChanges()`.
+```typescript
+export class PeekABoo implements OnInit {
+  constructor(private logger: LoggerService) { }
+
+  // implement OnInit's `ngOnInit` method
+  ngOnInit() { this.logIt(`OnInit`); }
+
+  logIt(msg: string) {
+    this.logger.log(`#${nextId++} ${msg}`);
+  }
+}
+```
+
+#### [ngDoCheck()](https://angular.io/guide/lifecycle-hooks#docheck)
+Detect and act upon changes that Angular can't or won't detect on its own.
+
+Called during every change detection run, immediately after `ngOnChanges()` and `ngOnInit()`.
+```typescript
+export class PeekABooComponent extends PeekABoo implements DoCheck {
+
+  // ...
+  
+  ngDoCheck() { 
+    this.logIt(`DoCheck`);
+  }
+  
+  // ...
+  
+}
+```
+
+#### [ngAfterContentInit()](https://angular.io/guide/lifecycle-hooks#aftercontent-hooks)
+Respond after Angular projects external content into the component's view.
+Called once after the first `ngDoCheck()`.
+```typescript
+export class PeekABooComponent extends PeekABoo implements AfterContentInit {
+
+  // ...
+
+  ngAfterContentInit() { this.logIt(`AfterContentInit`);  }
+  
+  // ...
+  
+}
```

Comment:
Could you delete these comments `// ...` ? They take too much space and it's pretty obvious that it's just a part of the component. Even the sole method code would suffice.